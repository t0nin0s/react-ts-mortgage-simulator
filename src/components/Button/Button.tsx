import * as React from 'react';
import * as FontAwesome from 'react-fontawesome';
const styles = require('./button.css');

interface ButtonProps {
  id: string;
  color?: string | boolean;
  icon: string;
  value: string;
  text: string;
  handleOnChange: (event: React.MouseEvent<HTMLInputElement>) => void;
  large?: boolean;
}

const MyButton: React.SFC<ButtonProps> = (props) => {
  let buttonStyle: string;
  if (props.large) {
    buttonStyle = [styles.button, styles.fullWidth].join(' ');
  } else {
    buttonStyle = [styles.button, styles.halfWidth].join(' ');
  }
  return (
    <button
      className={buttonStyle}
      id={props.id}
      value={props.value}
      onClick={props.handleOnChange}
    >
      <span
        className={styles.button__text}
      >
        {props.text}
      </span>
      <span className={styles.button__icon}>
        <FontAwesome
          value={props.value}
          name={props.icon}
          size="3x"
          style={{ color: props.color }}
        />
      </span>
    </button>
  );
};

export default MyButton;
