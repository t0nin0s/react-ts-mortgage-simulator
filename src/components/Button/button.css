:local(.button) {
  color: #A06FEF;
  font-weight: bold;
  background-color: white;
  border: 1px solid #ccc;
  height: 5rem;
  text-align: left;
  padding-left: 3rem;
  border-radius: 2rem;
  -moz-border-radius: 2rem;
  -webkit-border-radius: 2rem;
  margin: 0.5rem 0;
  z-index: 999;
}

:local(.button span) {
  pointer-events: none;
}

:local(.fullWidth) {
  width: 100%;
}

:local(.halfWidth) {
  width: 45%;
  margin: 0 2.5%;
}

:local(.button--selected) {
  color: red;
}

:local(.button__text) {
  float:left;
  font-weight:bold;
  line-height: 4rem;
}

:local(.button__icon) {
  height:4rem;
  display: inline;
  vertical-align: middle;
  float: right;
  padding-right: 1rem;
  z-index: 0;
}
