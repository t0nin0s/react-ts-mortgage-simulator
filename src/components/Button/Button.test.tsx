import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Button from './Button'

describe('SignupFormPersonal renders as expected', () => {
  it('should render as expected', () => {
    const component = shallow(
      <Button
        id='idComp'
        color='color'
        icon='icon'
        value='value'
        text='text'
        handleOnChange={jest.fn()}
      />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
