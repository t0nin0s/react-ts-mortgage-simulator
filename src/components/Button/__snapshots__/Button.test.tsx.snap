// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupFormPersonal renders as expected should render as expected 1`] = `
<button
  className=" "
  id="idComp"
  onClick={[Function]}
  value="value"
>
  <span>
    text
  </span>
  <span>
    <FontAwesome
      name="icon"
      size="3x"
      style={
        Object {
          "color": "color",
        }
      }
      value="value"
    />
  </span>
</button>
`;
